packages/next/src/build/utils.ts

```diff
@@ -1480,6 +1482,16 @@ export async function isPageStatic({
           {}
         )
 
+        if (nextConfigOutput === 'export') {
+          if (!appConfig.dynamic || appConfig.dynamic === 'auto') {
+            appConfig.dynamic = 'error'
+          } else if (appConfig.dynamic === 'force-dynamic') {
+            throw new Error(
+              'export const dynamic = "force-dynamic" cannot be used with "output: export". See more info here: https://nextjs.org/docs/advanced-features/static-html-export'
```

Comment:
Do we want to add page name or file name into error message? might be an improvement

Reply:
Fixed in [1aa1d58](https://github.com/vercel/next.js/pull/47022/commits/1aa1d580ee3e625f1d17c02ecc3a914b9b10df2e)